model-server-openapi/specifications/model-server.yaml

```diff
@@ -13,481 +13,700 @@ info:
+        default:
+          $ref: "#/components/responses/Problem"
     post:
-      operationId: postGenerateClientId
+      operationId: initializeRepository
+      summary: Creates a new repository
+      tags:
+        - repository management
+      parameters:
+        - name: useRoleIds
+          in: "query"
+          required: false
+          schema:
+            type: boolean
+            default: true
+      requestBody:
+        description: The new repo to create
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Repository"
       responses:
-        "200":
-          $ref: '#/components/responses/200'
-  /headers:
+        "201":
+          description: A new repository was created successfully
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Repository"
+        "400":
+          description: A repository of the same name already exists
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  /repositories/{repository}:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+
     get:
-      operationId: getHeaders
+      operationId: getRepository
+      tags:
+        - repository management
       responses:
         "200":
-          $ref: '#/components/responses/200'
-  /health:
+          description: The repository exists
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Repository"
+        "404":
+          $ref: "#/components/responses/NoSuchRepository"
+        default:
+          $ref: "#/components/responses/Problem"
+
+    delete:
+      operationId: deleteRepository
+      tags:
+        - repository management
+      responses:
+        "204":
+          description: Repository was deleted
+        "404":
+          $ref: "#/components/responses/NoSuchRepository"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  /repositories/{repository}/branches:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+
     get:
-      operationId: getHealth
+      operationId: getRepositoryBranches
+      tags:
+        - branch management
       responses:
         "200":
-          $ref: '#/components/responses/200'
-        "500":
-          $ref: '#/components/responses/500'
-  /v2/repositories:
+          description: Successful listing of all repository branches
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/BranchList"
+        "404":
+          $ref: "#/components/responses/NoSuchRepository"
+        default:
+          $ref: "#/components/responses/Problem"
+
+    post:
```

Comment:
`POST /repositories/{repository}/branches`, `PUT /repositories/{repository}/branches/{branch}` should additionally be able to create a branch from another branch name without knowing the version.